aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -329,6 +330,36 @@ public class AwsProperties implements Serializable {
   public static final String S3_DELETE_ENABLED = "s3.delete-enabled";
   public static final boolean S3_DELETE_ENABLED_DEFAULT = true;
 
+  /**
+   * Used by {@link S3FileIO} to tag objects when writing. To set, we can pass a catalog property.
+   * <p>
+   * For more details, see https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-tagging.html
+   * <p>
+   * Example: s3.write.table-uuid-tag-enabled=true
+   */
+  public static final String S3_WRITE_TABLE_UUID_TAG_ENABLED = "s3.write.table-uuid-tag-enabled";
```

Comment:
Using UUID as tag might be a bit tricker, because when creating table you don't know the table UUID yet when you create the table operation. So I would suggest we add table and namespace name tags first, and add UUID in a separated PR.

Reply:
Makes sense to me.